src/main/java/org/czekalski/petapiintegration/apiClient/v1/dto/AnimalsListDto.java

```diff
@@ -0,0 +1,46 @@
+package org.czekalski.petapiintegration.apiClient.v1.dto;
+
+
+import com.fasterxml.jackson.annotation.*;
+
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+
+public class AnimalsListDto {
+    private final   List<AnimalDto> animals;
+
+    private final  PaginationDto pagination;
+
+
+
+    private Map<String, Object> additionalProperties = new HashMap<String, Object>();
```

Comment:
Tak strzelam, że chciałeś tutaj wsypać wszystko, co się nie zmapowało na Twoje pola. W produkcyjnym kodzie się tak nie robi. Zazwyczaj dto są skrojone do potrzeb. Jeżeli JSON ma więcej pól niż DTO, to są one ignorowane. Ignorowanie pól to domyślne zachowanie springa, ale w samym jacksonie ustawia się to annotacją `@JsonIgnoreProperties(ignoreUnknown = true)` ustawioną na klasie

Reply:
Ja ogólnie bym się pozbył tego z wszystkich tych DTO

Reply:
Jest to też o tyle słabe, że nawet jak coś Ci się tam wrzuci, to będzie to `Object` i w sumie nawet jakby się znało klucz, to nie wiadomo, co za bardzo z tym zrobić. Czy to tylko tekst czy liczba czy data

Reply:
Ok, tutaj leciało to  _links